### PR TITLE
creates pages

### PR DESCRIPTION
**Pull request by sayeda-chattopadhyay**

### CODE REVIEW

File: src/CartContext.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    } else {
+      setCartProducts(
+        cartProducts.map(
+          (product) =>
+            product.id === id // if condition
+              ? { ...product, quantity: product.quantity - 1 } // if statement is true
+              : product // if statement is false
+        )
+      );
+    }
```
Comment (Fermain) [inline src/CartContext.jsx:97]: I think you need to update local storage here.

File: src/CartContext.jsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  // delete from cart
+
+  function deleteFromCart(id) {
+    // [] if an object meets a condition, add the object to array
+    setCartProducts((cartProducts) =>
+      cartProducts.filter((currentProduct) => {
+        return currentProduct.id !== id;
+      })
+    );
```
Comment (Fermain) [inline src/CartContext.jsx:109]: Same here

File: src/api/Product.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import ProductCard from "../components/card/ProductCard";
+
+const Product = () => {
+  const [product, setProduct] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  let { id } = useParams();
+
+  useEffect(() => {
+    async function getData(url) {
```
Comment (Fermain) [inline src/api/Product.jsx:13]: Why don't you use the ApiHook here?

File: src/api/Products.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (isError) {
+    return <div>Error</div>;
+  }
+
+  const products = data;
+  console.log(products);
+
+  return (
+    <>
+      <StyledSearch>
```
Comment (Fermain) [inline src/api/Products.jsx:27]: This could be it's own component

File: src/components/button/Button.jsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Button = () => {
+  return (
+    <div>Button</div>
```
Comment (Fermain) [inline src/components/button/Button.jsx:5]: Button using a div tag?

File: src/pages/ProductPage.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import Product from "../api/Product";
+import BreadCrumbs from "../components/breadcrumbs/BreadCrumbs";
+
+const ProductPage = () => {
+  return (
+    <div className="product-page">
```
Comment (Fermain) [inline src/pages/ProductPage.jsx:6]: Generally speaking its better to fetch the data in the page and pass it to components as a prop. This makes testing a lot easier in future.

**Review by Fermain — CHANGES_REQUESTED** [review]

File: src/CartContext.jsx (hunk 1 above)
Comment (sayeda-chattopadhyay) [reply src/CartContext.jsx:97]: updated

File: src/CartContext.jsx (hunk 2 above)
Comment (sayeda-chattopadhyay) [reply src/CartContext.jsx:109]: done

File: src/components/button/Button.jsx (hunk 5 above)
Comment (sayeda-chattopadhyay) [reply src/components/button/Button.jsx:5]: Ok. I will change it.

File: src/pages/ProductPage.jsx (hunk 6 above)
Comment (sayeda-chattopadhyay) [reply src/pages/ProductPage.jsx:6]: Ok. I will do that. Thanks for the information.

**Pull request merged**